zest/releaser/pypi.py

```diff
@@ -252,6 +278,62 @@ def distutils_servers(self):
         # The servers all need to have a section in the config file.
         return [server for server in index_servers if self.config.has_section(server)]
 
+
+class PyprojectTomlConfig(BaseConfig):
+    """Wrapper around the pyproject.toml file if available.
+
+    This is for optional zest.releaser-specific settings::
+
+        [zest-releaser]
+        python-file-with-version = "reinout/maurits.py"
+
+
+    """
+
+    config_filename = PYPROJECTTOML_CONFIG_FILE
+
+    def __init__(self):
+        """Grab the configuration (overridable for test purposes)"""
+        # If there is a pyproject.toml in the package, parse it
+        if not os.path.exists(self.config_filename):
+            self.config = None
+            return
+        with open(self.config_filename, "rb") as tomlconfig:
+            self.config = tomllib.load(tomlconfig)
+
+    def zest_releaser_config(self):
+        if self.config is None:
+            return None
+        try:
+            result = self.config["tool"]["zest-releaser"]
+        except KeyError:
+            return None
+        return result
+
+
+class ZestReleaserConfig:
+    hooks_filename = None
+
+    def load_configs(self, pypirc_config_filename=DIST_CONFIG_FILE):
+        setup_config = SetupConfig()
+        pypi_config = PypiConfig(config_filename=pypirc_config_filename)
+        pyproject_config = PyprojectTomlConfig()
+        combined_config = {}
+        # overwrite any duplicate keys in the following order:
+        for config in [setup_config, pypi_config, pyproject_config]:
+            if config.zest_releaser_config() is not None:
+                zest_config = config.zest_releaser_config()
+                assert isinstance(zest_config, dict)
+                combined_config.update(zest_config)
+
+                # store which config file contained entrypoint hooks
+                if any([x for x in zest_config.keys() if x.lower().startswith(("prereleaser.", "releaser.", "postreleaser."))]):
```

Comment:
I'm suspecting a 'black' is in order here?

I'm also suspecting I have to do some pre-commit setup for zest.releaser :-)